packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -118,22 +122,27 @@ export class PendingStateManager implements IDisposable {
 		return this.pendingMessagesCount !== 0;
 	}
 
-	public getLocalState(): IPendingLocalState | undefined {
+	public getLocalState(snapshotSequenceNumber?: number): IPendingLocalState {
 		assert(
 			this.initialMessages.isEmpty(),
 			0x2e9 /* "Must call getLocalState() after applying initial states" */,
 		);
-		if (!this.pendingMessages.isEmpty()) {
-			return {
-				pendingStates: [...this.savedOps, ...this.pendingMessages.toArray()].map(
-					(message) => {
-						// delete localOpMetadata since it may not be serializable
-						// and will be regenerated by applyStashedOp()
-						return { ...message, localOpMetadata: undefined };
-					},
-				),
-			};
-		}
+		const newSavedOps = [...this.savedOps].filter(
+			(message) => message.referenceSequenceNumber >= (snapshotSequenceNumber ?? 0),
+		);
+		return {
+			pendingStates: [...newSavedOps, ...this.pendingMessages.toArray()].map((message) => {
+				if (
+					snapshotSequenceNumber &&
```

Comment:
Please use snapshotSequenceNumber  !== undefined.
Zero is falsey, but it's a valid value for a reference sequence number of a summary!